provision: Added msExchOmaAdminWirelessEnable attribute to user

javierag opened the pull request:
This attribute is needed to manage ActiveSync ACL per user.

Since this change is transparent to end-user there is no CHANGES.md line

sixstone-qq commented on python/openchange/migration/directory.py:
```diff
@@ -26,6 +26,24 @@
+        return get_local_samdb(names, lp, creds)
+
+
+def _get_all_enabled_users(db, **kwargs):
```
I'll move this function within Migration AddActiveSyncAttributes class as it is exclusive.

blaxter replied:
+1

javierag replied:
I wanted to leave them outside as utility functions for future provisions, if you two dont like the idea I will inline them .

sixstone-qq commented on python/openchange/migration/directory.py:
```diff
@@ -26,6 +26,24 @@
 import sys
 
 
+def _get_local_samdb(**kwargs):
```
I see no benefit of refactoring here.

blaxter commented on python/openchange/migration/directory.py:
```diff
@@ -235,3 +253,45 @@ def unapply(cls, cur, **kwargs):
+
+    @classmethod
+    def apply(cls, cur, **kwargs):
+        db = _get_local_samdb(**kwargs)
```
`db = get_local_samdb(kwargs['lp'], kwargs['creds'], kwargs['names'])`

javierag commented:
Remember to add msExchOmaAdminWirelessEnable attributes to list of attributes to delete from user when t https://github.com/openchange/openchange/pull/338  is merged

sixstone-qq commented on python/openchange/migration/directory.py:
```diff
@@ -235,3 +235,60 @@ def unapply(cls, cur, **kwargs):
+        from openchange.provision import get_local_samdb
+        lp = kwargs['lp']
+        creds = kwargs['creds']
+        names = kwargs['names']
```
The above three lines are unnecessary and IMHO it obfuscates the code, using kwargs['lp'] and such is more concise and clearer from my POV.

javierag replied:
I rather like to unroll the parameters. Also this has been done in the other migrations of the same file.

sixstone-qq commented:
@javierag if you want to merge the delete user feature, I'd rather separate it as I don't see any discussion on that topic. Once this is done, I will add that commit here, rebase your code onto it and, finally, add new code in another commit or the same one for deleting the user use case.

javierag commented:
No probel, the commentary was a note for myself to add the property to the delete user attributes when this pr is merged.

exekias commented on python/openchange/migration/directory.py:
```diff
@@ -235,3 +235,60 @@ def unapply(cls, cur, **kwargs):
+        ldb_filter = "(&(objectClass=user)(proxyAddresses=*))"
+        return db.search(base=base_dn,
+                         scope=ldb.SCOPE_SUBTREE,
+                         expression=ldb_filter)
```
Is the migration needed? can't we understand that the missing atribute means 0?

javierag replied:
No because SOGO's modulesConstraint code fails the check if the attribute does not exists

exekias replied:
what do you mean with failed? it crashes or sets to NO?

javierag replied:
The check return false

javierag commented:
In the newuser and newgroup ldifs I have changed the 'add' operation for attributes to a replace operation.  The reason is that I had something else adding msExOmaAdminWirelessEnable before user provision.
Thinking about it, we want a replace operation to assure the user/group has the correct attributes, just adding a existent attr would trigger error if single-valued or had complications if multi-valued.
So I changed all to replace.

sixstone-qq commented:
@javierag if your reasoning is correct, I will provide another PR with the 'replace' thing or, at least, another commit. This thing has nothing to do with adding msExOmaAdminWirelessEnable attribute...

javierag commented:
Yes, but i need the replace to wrok wth the msExOmaAdminWirelessEnable

javierag commented:
I have rebased to master and I have added msExchOmaAdminWirelessEnable attribute to the list of attributes to delete

blaxter commented:
:+1:
